build/circle-test.sh

```diff
@@ -1,25 +1,82 @@
 #!/bin/bash
 
-set -e
+set -euo pipefail
 
-outdir="${TMPDIR}"
-if [ "${CIRCLE_ARTIFACTS}" != "" ]; then
+if [ -z "${CIRCLE_ARTIFACTS-}" ]; then
   outdir="${CIRCLE_ARTIFACTS}"
+elif [ -z "${TMPDIR-}" ]; then
+  outdir="${TMPDIR}"
+else
+  outdir=""
 fi
 
 builder=$(dirname $0)/builder.sh
 
+prepare_artifacts() {
+  # Translate the log output to xml to integrate with CircleCI
+  # better.
+  if [ -z "${CIRCLE_TEST_REPORTS-}" ]; then
+    if [ -f "${outdir}/test.log" ]; then
+      mkdir -p "${CIRCLE_TEST_REPORTS}/go"
+      ${builder} go2xunit < "${outdir}/test.log" \
+        > "${CIRCLE_TEST_REPORTS}/go/test.xml"
+    fi
+    if [ -f "${outdir}/testrace.log" ]; then
+      mkdir -p "${CIRCLE_TEST_REPORTS}/race"
+      ${builder} go2xunit < "${outdir}/testrace.log" \
+        > "${CIRCLE_TEST_REPORTS}/race/testrace.xml"
+    fi
+    if [ -f "${outdir}/acceptance.log" ]; then
+      mkdir -p "${CIRCLE_TEST_REPORTS}/race"
```

Comment:
We should make a new directory for acceptance tests instead of reusing `race`; IIRC the directory name (and not the filename) is used in circleci's reporting.